tests/kernel/fp_sharing/src/float_context.h

```diff
@@ -88,6 +88,24 @@ struct fp_non_volatile_register_set {
 #define SIZEOF_FP_NON_VOLATILE_REGISTER_SET \
 	sizeof(struct fp_non_volatile_register_set)
 
+#elif defined(CONFIG_CPU_ARCV2)
+
+struct fp_volatile_register_set {
+#ifdef CONFIG_FP_FPU_DA
+	u32_t dpfp2h;
+	u32_t dpfp2l;
+	u32_t dpfp1h;
+	u32_t dpfp1l;
+#endif
```

Comment:
How about r58, r59, and FPU status?  It's just a test, but without the single precision FP registers included and 'trashed' only the DA variant gets tested. 

I didn't get the difference between 'volatile' and 'non_volatile' register set here, btw. Not relevant for ARC?

Reply:
for r58 r59,  as they are shared with hardware divider, see instructions divu,  in this test,  r58 r59 will be polluted by divu instructions in z_tick_get_32.  So I just remove them

For ctrl and status, they are aux regs, and not 32-bit r/w,  don't match the mechanisam in the fp_sharing tests., so remove them

For fp_sharing test, i think dpfpxx regs are enough to prove floating point context are saved and restored during context switch.

Reply:
I'd prefer the test to be more explicit in changing all possible FPU context, instead of relying on implicit changes due to a divu call (which might be avoided/changed during maintenance updates of this code).

Also, if r58/r59 can be used by non FP code, shouldn't they be always saved/restored instead of being part of the lazy 'FP_SHARING' context?